File: DSA_cpp/search/binary_search.cpp
//
// Created by 吳亭緯 on 2018-12-22.
//

#include <iostream>
#include <fstream>
#include <string>

using namespace std;

ifstream f("data.in");
ofstream g("data.out");

int A[1000], numberElements;

int BinarySearch(int X){
    int Left=1, Right=numberElements, mid;

    while(Left <= Right){
        mid = (Left+Right) / 2;
        if(X==A[mid]) return mid;
        else if(X<A[mid]) Right=mid-1;
        else Left=mid+1;
    }
    return -1;
}

int main(){

    int answer;
    f>>numberElements;
    for(int i=0; i<numberElements; ++i){
        f>>A[i];
    }
    answer = BinarySearch(5);
    cout << answer;
    g.write(answer);
}

File: DSA_python/binary_search.py
def binary_search(arr,start,end,hkey):
	if start > end:
		return -1
	mid = start + (end - start) / 2
	if arr[mid] > hkey:
		return binary_search(arr, start, mid - 1, hkey)
	if arr[mid] < hkey:
		return binary_search(arr, mid + 1, end, hkey)
	return mid

def binary_search(arr, start, end, hkey):
	while start <= end:
		mid = start + (end - start) / 2
		if arr[mid] < hkey:
			start = mid + 1
		elif arr[mid] > hkey:
			end = mid - 1
		else:
			return mid
     return -1